cdap-api/src/main/java/io/cdap/cdap/api/plugin/PluginGroupConfig.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.api.plugin;
+
+import io.cdap.cdap.api.Config;
+
+/**
+ * Config that can be defined as a field inside {@link PluginConfig} to represent a collection of configs.
+ * When the plugin is deployed, the configs inside this class will be inspected and represented in the same way as
+ * {@link PluginPropertyField}.
+ * The {@link PluginPropertyField} for this field will contain a collection of property names about configs inside
+ * this class.
+ */
+public class PluginGroupConfig extends Config {
```

Comment:
Config is an application config and shouldn't be extended.

I'm wondering whether we need this class, could the contract be that the field should just extend the existing PluginConfig class? Wouldn't we need some of PluginConfig's methods for checking if a property contains a macro anyway?

Reply:
Initially the implementation is to have all the macro fields in the parent class to check, on a second thought to have it in the child class is better to implement a plugin, so removed this class and forced to use PluginConfig, it adds a bit complication on set the field for the nested config though